libs/dyn/yamlsaver/saver.go

```diff
@@ -13,7 +13,23 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-func SaveAsYAML(data any, filename string, force bool) error {
+type saver struct {
+	nodesWithStyle map[string]yaml.Style
+}
+
+func NewSaver() *saver {
+	return &saver{
+		nodesWithStyle: make(map[string]yaml.Style),
```

Comment:
It should be fine to zero-initialize; lookups in a nil map are ok, assignment panics.